packages/full-stack-tests/src/hierarchy-builder/grouping/BaseClassGrouping.test.ts

```diff
@@ -0,0 +1,764 @@
+
+import { PhysicalPartition, Subject } from "@itwin/core-backend";
+import { IModel } from "@itwin/core-common";
+import { IModelConnection } from "@itwin/core-frontend";
+import { SchemaContext } from "@itwin/ecschema-metadata";
+import { ECSchemaRpcLocater } from "@itwin/ecschema-rpcinterface-common";
+import { createECSqlQueryExecutor, createMetadataProvider } from "@itwin/presentation-core-interop";
+import { HierarchyNode, HierarchyProvider, IHierarchyLevelDefinitionsFactory, NodeSelectClauseFactory } from "@itwin/presentation-hierarchy-builder";
+import { buildIModel, insertPhysicalPartition, insertSubject } from "../../IModelUtils";
+import { initialize, terminate } from "../../IntegrationTests";
+import { NodeValidators, validateHierarchy } from "../HierarchyValidation";
+
+describe("Stateless hierarchy builder", () => {
+  describe("BaseClassGrouping", () => {
+    let selectClauseFactory: NodeSelectClauseFactory;
+    let subjectClassName: string;
+    let physicalPartitionClassName: string;
+
+    before(async function () {
+      await initialize();
+      subjectClassName = Subject.classFullName.replace(":", ".");
+      physicalPartitionClassName = PhysicalPartition.classFullName.replace(":", ".");
+      selectClauseFactory = new NodeSelectClauseFactory();
+    });
+
+    after(async () => {
+      await terminate();
+    });
+
+    function createProvider(props: { imodel: IModelConnection; hierarchy: IHierarchyLevelDefinitionsFactory }) {
+      const { imodel, hierarchy } = props;
+      const schemas = new SchemaContext();
+      schemas.addLocater(new ECSchemaRpcLocater(imodel.getRpcProps()));
+      const metadataProvider = createMetadataProvider(schemas);
+      return new HierarchyProvider({
+        metadataProvider,
+        hierarchyDefinition: hierarchy,
+        queryExecutor: createECSqlQueryExecutor(imodel),
+      });
+    }
+    describe("does not group", () => {
+      let sharedIModel: IModelConnection;
+      let sharedKeys: any;
+      beforeEach(async function () {
+        const { imodel, ...keys } = await buildIModel(this, async (builder) => {
+          const rootSubject = { className: subjectClassName, id: IModel.rootSubjectId };
+          const childSubject1 = insertSubject({ builder, codeValue: "A1", parentId: rootSubject.id });
+          const childSubject2 = insertSubject({ builder, codeValue: "A2", parentId: rootSubject.id });
+          const childPartition3 = insertPhysicalPartition({ builder, codeValue: "B3", parentId: rootSubject.id });
+          const childPartition4 = insertPhysicalPartition({ builder, codeValue: "B4", parentId: rootSubject.id });
+          const childPartition5 = insertPhysicalPartition({ builder, codeValue: "B5", parentId: rootSubject.id });
+          return { rootSubject, childSubject1, childSubject2, childPartition3, childPartition4, childPartition5 };
+        });
+        sharedKeys = keys;
+        sharedIModel = imodel;
+      });
```

Comment:
I think for these tests you could use an empty iModel (which has 1 root `Subject`).

Reply:
Using an empty iModel for these tests:
- doesn't create grouping nodes if provided classes aren't base for node class
- doesn't create grouping nodes if provided classes aren't of entity or relationship type
- creates grouping nodes if provided class is base for node class